llarp/config/config.cpp

```diff
@@ -1132,10 +1132,12 @@ namespace llarp
     if (ec)
       throw std::runtime_error(stringify("filesystem error: ", ec));
 
+    const auto parent = confFile.parent_path();
+
     // create parent dir if it doesn't exist
-    if (not fs::exists(confFile.parent_path(), ec))
+    if ((not parent.empty()) and (not fs::exists(parent, ec)))
```

Comment:
Since we're passing `basedir` in here as `defaultDataDir`, it's weird to use `parent_path` again here.

Reply:
Also the parameter `defaultDataDir` here should probably be renamed; suggestion: `dataDir`.